source/use-cases/home-resolver.rst

```diff
@@ -1,6 +1,129 @@
+
+Setting up your own DNS resolver for your entire home network requires a couple of things. Namely, a recursive DNS resolver (who knew!), and a dedicated machine where the resolver runs, which is always on and accessible to the entire network. This can be as simple as a Raspberry Pi connected to your home router or any other machine that you is always online!
+
+Because of the variaty of machines that Unbound can run on we cannot create a comprehensive tutorial for all possible options. For the extent of this tutorial we will Ubuntu as a stepping stone which you could use for other machines. For this tutorial Ubuntu 20.04.1 LTS was used.
+
+While you could download the code and build it yourself if wou wanted, it can be as simple as running:
+::
+	apt install unbound
+
+This gives you a full, compiled, and running version of Unbound which behaves as a caching recursive DNS resolver out of the box for the local machine.
+
+Do note that the current setup is only reachable on this machine.
+
+Testing the server locally
 --------------------------
 
-Using Unbound as the resolver for a home network.
+To verify that the server works correctly it’s a good idea to test it before committing the entire network to it. Luckily we can test this on the machine that you installed Unbound on (local) and from any other machine (remote)  that will be using the resolver after we expose Unbound to the network.
+
+The command for local testing is:
+::
+	dig example.com @127.0.0.1
+
+So we tell the dig tool to look up the IP address for example.com, and to ask this information to the server running at the ip address 127.0.0.1, which is where our Unbound machine is running.
+
+
+Setting up for a single machine
+-------------------------------
+
+Now that we have configured and tested our Unbound server, we can tell our machine to use by default. We do this at:
+::
+	sudo nano /etc/systemd/resolved.conf
+Here we change the entries under [Resolve] to use our own instance running at 127.0.0.1 
+::
+	sudo systemctl restart systemd-resolved
+
+We then need to stop the currently running pre-installed resolver:
+::
+	sudo systemctl disable systemd-resolved.service
+	sudo systemctl stop systemd-resolved
+
+And now we can then start using our new configuration:
+::
+	  sudo service network-manager restart
+
+And as a quick test a *dig* without specifying our Unbound server should give the same result as specifying it (with the *@127.0.0.1*)!
+::
+	dig example.com
+
+Setting up for the rest of the network
+--------------------------------------
+
+While we current have a working instance of unbound, we need it to be reachable from within our entire network. With that comes the headache of dealing with IP addresses. It’s likely that your home router distributed local IP addresses to your devices. If this is the case (i.e. you didn’t change it by hand), the ranges should be between[`RFC1918<http://tools.ietf.org/html/rfc1918>`]:
```

Comment:
"While we current" -> "While we currently"
"distributed local IP addresses to your devices" -> "assigned local IP addresses vai DHCP to your devices" ?
"the ranges should be between" -> " The IP addresses should be in one of the following ranges" ?